loopd/migration.go

```diff
@@ -0,0 +1,87 @@
+package loopd
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+
+	"github.com/lightninglabs/lndclient"
+	"github.com/lightninglabs/loop/loopdb"
+	"github.com/lightningnetwork/lnd/lnrpc"
+)
+
+// migrateBoltdb migrates the boltdb to sqlite.
+func migrateBoltdb(ctx context.Context, cfg *Config) error {
+
+	// First get the chain params.
+	chainParams, err := lndclient.Network(cfg.Network).ChainParams()
+	if err != nil {
+		return err
+	}
+
+	// First open the bolt db.
+	boltdb, err := loopdb.NewBoltSwapStore(cfg.DataDir, chainParams)
+	if err != nil {
+		return err
+	}
+
+	// Now open the sqlite db.
+	sqliteDb, err := loopdb.NewSqliteStore(
+		&loopdb.SqliteConfig{
+			DatabaseFileName: filepath.Join(cfg.DataDir, "loop.sqlite"),
+		},
+		chainParams,
+	)
+	if err != nil {
+		return err
+	}
+
+	// Create a new migrator manager.
+	migrator := loopdb.NewMigratorManager(boltdb, sqliteDb)
+
+	// Run the migration.
+	err = migrator.RunMigrations(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Close both dbs.
+	err = boltdb.Close()
+	if err != nil {
+		return err
+	}
+
+	err = sqliteDb.Close()
+	if err != nil {
+		return err
+	}
+
+	// If the migration was successfull we'll rename the bolt db to
+	// loop.db.bk.
+	err = os.Rename(
```

Comment:
is keeping the backup what usually happens with migrations?

what if db was super big and keeping a duplicate causes problems?
(e.g a user with many swaps like muun)
is the following a possible scenario:
- muun upgrades & triggers migration
- migration fails due to running out of space
- now loop falls back to using boltdb (?)

Reply:
I agree with @sputn1ck, it's important to keep the old DB around. It's also likely not large enough to cause any issues regarding storage space and if it is, then the migration above will already fail since this is just a rename.